### PR TITLE
Feat/searchbar

### PR DESCRIPTION
**Pull request by leoguzi**
- A search on the server is triggered everytime a new char is typed and the results are shown on the layout;
- The search is triggered only when the user typed at least 3 chars and stoped for at least 300ms;
- The profiles that are followed by the user are shown in the top of the list;
- When the user click on the result, gets redirected to the respective profile;

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -22,6 +21,7 @@ export default function Header() {
         <Link to="/timeline">
           <h1>linkr</h1>
         </Link>
+        {window.innerWidth > 1000 ? <SearchUser /> : ""}
```
Comment (Calil-Silva) [inline src/components/Header.js:24]: Leo! Estou adotando, quando lembro, de fazer a seguinte sintax:
        {window.innerWidth > 1000 && <SearchUser />}
Assim evitamos de colocar "".

File: src/components/SearchUser.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          <img src={user.avatar} alt={user.username + "'s avatar"} />
+          <span>{user.username}</span>
+          {user.isFollowingLoggedUser ? (
+            <TextContainer>
+              <Dot />
+              <span>following</span>
+            </TextContainer>
+          ) : (
+            ""
+          )}
```
Comment (Calil-Silva) [inline src/components/SearchUser.js:36]: Aqui também daria pra usar o && para não voltar o "".

File: src/components/SearchUser.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    );
+  }
+
+  return (
+    <>
+      {searchField.length > 2 ? (
+        <BlockContent onClick={() => setSearchField("")} />
+      ) : (
+        ""
+      )}
```
Comment (Calil-Silva) [inline src/components/SearchUser.js:48]: Aqui também daria pra usar o && para não voltar o "".

**Comment by Calil-Silva** [comment]
In general, it looks all right. All requirements seems to be running on Vercel.

Tips: We can use "&&" to "||" instead of returning "" in JSX. Something like:

```
// {true && true}, instead of {true ? x : ""}
// {true || false}, instead of {true ? x : y} 
```

**Review by calilrenner — COMMENTED** [review]
In general, it looks all right. All requirements seems to be running on Vercel.

Tips: We can use "&&" or "||" instead of returning "" in JSX. Something like:

```
// {true && true}, instead of {true ? x : ""}
// {true || false}, instead of {true ? x : y}
```

File: src/components/Header.js (hunk 1 above)
Comment (sidneyouteiro) [inline src/components/Header.js]: `{window.innerWidth > 1000 && <SearchUser />}`

File: src/components/SearchUser.js (hunk 2 above)
Comment (sidneyouteiro) [inline src/components/SearchUser.js]:
```
{user.isFollowingLoggedUser && (
            <TextContainer>
              <Dot />
              <span>following</span>
            </TextContainer>
          )}
```

File: src/components/SearchUser.js (hunk 3 above)
Comment (sidneyouteiro) [inline src/components/SearchUser.js]:
```
{searchField.length > 2 && <BlockContent onClick={() => setSearchField("")} />}
```

File: src/components/SearchUser.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+        <SearchIcon />
+        {searchField.length > 2 && searchResult.length === 0 ? (
+          <a>
+            <Result>
+              <span>Nenhum resultado...</span>
+            </Result>
+          </a>
+        ) : (
+          ""
+        )}
```
Comment (sidneyouteiro) [inline src/components/SearchUser.js]:
```
{searchField.length > 2 && !searchResult.length && (
          <a>
            <Result>
              <span>Nenhum resultado...</span>
            </Result>
          </a>
        )}
```

**Review by calilrenner — APPROVED** [review]
In general, it looks all right. All requirements seems to be running on Vercel.

File: src/pages/Hashtag.js [hunk 5]
```diff
@@ -22,6 +23,7 @@ export default function Hashtag() {
     <>
       <Header />
       <Main>
+        {window.innerWidth < 1000 ? <SearchUser /> : ""}
```
Comment (sidneyouteiro) [inline src/pages/Hashtag.js]: `{window.innerWidth < 1000 && <SearchUser />}`

File: src/pages/MyLikes.js [hunk 6]
```diff
@@ -19,6 +20,7 @@ export default function MyLikes() {
     <>
       <Header />
       <Main>
+        {window.innerWidth < 1000 ? <SearchUser /> : ""}
```
Comment (sidneyouteiro) [inline src/pages/MyLikes.js]: `{window.innerWidth < 1000 && <SearchUser />}`

File: src/pages/MyPosts.js [hunk 7]
```diff
@@ -22,6 +23,7 @@ export default function MyPosts() {
     <Main>
       <div>
         <Header />
+        {window.innerWidth < 1000 ? <SearchUser /> : ""}
```
Comment (sidneyouteiro) [inline src/pages/MyPosts.js]: `{window.innerWidth < 1000 && <SearchUser />}`

File: src/pages/Timeline.js [hunk 8]
```diff
@@ -53,10 +54,11 @@ export default function Timeline() {
         <Main>
           <div>
             <Header />
+            {window.innerWidth < 1000 ? <SearchUser /> : ""}
```
Comment (sidneyouteiro) [inline src/pages/Timeline.js]: `{window.innerWidth < 1000 && <SearchUser />}`

File: src/pages/UserPosts.js [hunk 9]
```diff
@@ -21,54 +27,56 @@ export default function UserPosts() {
...
+    [following, id] // eslint-disable-line react-hooks/exhaustive-deps
   );
 
   useEffect(
     () => {
-      setFollowsId(follows.map(user => user.id))
+      setFollowsId(follows.map((user) => user.id));
       // setFollowing(followsId.includes(parseInt(id)) || false)
     },
     [follows] // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (sidneyouteiro) [inline src/pages/UserPosts.js]: Remover comentário na linha 39, caso dê warning no useEffect me mande msg com o warning que eu já tenho uma ideia de como resolver

File: src/pages/UserPosts.js [hunk 10]
```diff
@@ -21,54 +27,56 @@ export default function UserPosts() {
...
+        alert("Não foi possível executar a operação.");
+        setDisabled(false);
+      });
   }
 
   return (
     <>
       <Header />
       <Main>
+        {window.innerWidth < 1000 ? <SearchUser /> : ""}
```
Comment (sidneyouteiro) [inline src/pages/UserPosts.js]: `{window.innerWidth < 1000 && <SearchUser />}`

**Review by sidneyouteiro — CHANGES_REQUESTED** [review]

**Review by sidneyouteiro — APPROVED** [review]

**Pull request merged**